tests/sentry/hybridcloud/models/test_outbox.py

```diff
@@ -324,55 +404,161 @@ def test_skip_shards(self) -> None:
+        ):
+
+            def process_with_error() -> None:
+                using = router.db_for_write(RegionOutbox)
+                with connections[using].cursor() as cursor:
+                    cursor.execute("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE")
+                    cursor.close()
+
+                # First, get the coalesced messages
+                coalesced_id = None
+                with transaction.atomic(using=using):
+                    coalesced = (
+                        outbox1.select_coalesced_messages().select_for_update(nowait=True).first()
+                    )
+                    if coalesced:
+                        coalesced_id = coalesced.id
+
+                if coalesced_id:
+                    # Process outside of transaction
+                    coalesced = RegionOutbox.objects.get(id=coalesced_id)
+                    coalesced.send_signal()
+                    # Delete inside transaction
+                    with transaction.atomic(using=using):
+                        RegionOutbox.objects.filter(id=coalesced_id).delete()
+
+            process_with_error()
+
+        # Verify that some messages were processed but not all
+        assert RegionOutbox.objects.count() == 2
+
+    def test_process_coalesced_large_batch(self) -> None:
+        """Test processing a large number of coalesced messages"""
+        num_messages = 200  # More than batch size of 50
+        outboxes = []
+
+        for i in range(num_messages):
+            outbox = OrganizationMember(id=i + 1, organization_id=3, user_id=1).outbox_for_update()
+            outboxes.append(outbox)
+
+        with outbox_context(flush=False):
+            for outbox in outboxes:
+                outbox.save()
+
+        # Verify initial state
+        assert RegionOutbox.objects.count() == num_messages
+
+        # Process all messages
+        def process_large_batch() -> None:
+            using = router.db_for_write(RegionOutbox)
+            with connections[using].cursor() as cursor:
+                cursor.execute("SET TRANSACTION ISOLATION LEVEL SERIALIZABLE")
+                cursor.close()
+
+            while RegionOutbox.objects.exists():
+                # First, get all coalesced messages
+                coalesced_ids = []
+                with transaction.atomic(using=using):
+                    coalesced = (
+                        RegionOutbox.objects.filter(category=outboxes[0].category)
+                        .select_for_update(nowait=True)
+                        .values_list("id", flat=True)
+                    )
+                    coalesced_ids = list(coalesced)
+
+                if not coalesced_ids:
+                    break
+
+                # Process outside of transaction
+                for coalesced_id in coalesced_ids:
+                    coalesced_obj = RegionOutbox.objects.get(id=coalesced_id)
+                    coalesced_obj.send_signal()
```

Comment:
Why is the signal being triggered for each coalesced message? We should only be triggering the signal once for the entire group.